lib/ace/keyboard/vim.js

```diff
@@ -489,8 +489,12 @@ define(function(require, exports, module) {
   };
   this.scrollInfo = function() { return 0; };
   this.scrollIntoView = function(pos, margin) {
-    if (pos)
-      this.ace.renderer.scrollCursorIntoView(toAcePos(pos), null, margin);
+    if (pos) {
+      var renderer = this.ace.renderer;
+      var viewMargin = { "top": 0, "bottom": margin };
+      renderer.scrollCursorIntoView(toAcePos(pos),
+        (renderer.lineHeight * 2) / renderer.$size.scrollerHeight, viewMargin);
```

Comment:
Looks good! Only one small issue.
This aligns cursor to the top of the viewport. But that won't work well for backwards search, and I think Vim always scrolls it to the center, so 0.5 may be better here.